examples/electron_mpi.py

```diff
@@ -0,0 +1,179 @@
+@click.option('-o', '--outer-radii', help='The outer radii (comma separated)',
+              required=True)
+@click.option('-f', '--output-file', help='The output HDF5 file to write',
+              default='electron_counted_data.h5')
+@click.option('-g', '--generate-sparse', is_flag=True,
+              help='Generate and save sparse STEM image')
+def main(files, dark_file, center, inner_radii, outer_radii, output_file,
+         generate_sparse):
+    center = center.split(',')
+    if len(center) != 2:
+        msg = 'Center must be of the form: center_x,center_y.'
+        raise click.ClickException(msg)
+
+    center_x, center_y = [int(x) for x in center]
+
+    inner_radii = inner_radii.split(',')
+    outer_radii = outer_radii.split(',')
+
+    if len(inner_radii) != len(outer_radii):
+        msg = 'Number of inner and outer radii must match'
+        raise click.ClickException(msg)
+
+    inner_radii = [int(x) for x in inner_radii]
+    outer_radii = [int(x) for x in outer_radii]
+
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    world_size = comm.Get_size()
+
+    if (world_size > len(files)):
+        if rank == 0:
+            print('Error: number of MPI processes,', world_size, ', exceeds',
+                  'the number of files:', len(files))
+        return
+
+    comm.Barrier()
+    start = MPI.Wtime()
+
+    # Every process will do the dark field reference average for now
+    reader = io.reader(dark_file, version=io.FileVersion.VERSION3)
+    dark = image.calculate_average(reader)
+
+    # Split up the files among processes
+    files = get_files(files)
+
+    # Create local electron count
+    reader = io.reader(files, version=io.FileVersion.VERSION3)
+    electron_counted_data = image.electron_count(reader, dark, verbose=True)
+    local_frame_events = electron_counted_data.data
+
+    # Now reduce to root
+    global_frame_events = reduce_to_root_method1(local_frame_events)
+    # global_frame_events = reduce_to_root_method2(local_frame_events)
+
+    comm.Barrier()
+    end = MPI.Wtime()
+
+    if rank == 0:
+        print('time: %s' % (end - start))
+
+    if rank == 0:
+        # Write out the HDF5 file
+        scan_width = electron_counted_data.scan_width
+        scan_height = electron_counted_data.scan_height
+        frame_width = electron_counted_data.frame_width
+        frame_height = electron_counted_data.frame_height
+
+        io.save_electron_counts(output_file, global_frame_events, scan_width,
+                                scan_height, frame_width, frame_height)
+
+        # Save out the electron counted image
+        img = np.zeros((576, 576))
+
+        # Just sum the events for now
+        for frame in global_frame_events:
+            for pos in frame:
+                row = pos // 576
+                column = pos % 576
+                img[row][column] += 1
+
+        fig, ax = plt.subplots(figsize=(12, 12))
+        ax.matshow(img)
+        plt.savefig('electron.png', dpi=300)
```

Comment:
We should make this optional as well or just remove it.

Reply:
Done.